fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -319,14 +318,24 @@ private int getChildrenLimit() {
     @DELETE
     @Timed
     public Response deleteObject() {
-        evaluateRequestPreconditions(request, servletResponse, resource(), session);
+        final FedoraResource resource = resource();
+        if (resource instanceof ContainerImpl) {
```

Comment:
Unfortunately, this brings a modeshape implementation class into `fcrepo-http-api`.   `ContainerImpl` should be removed and replace with something else.  I think  plain ol' `org.fcrepo.kernel.api.models.Container` would work.

Reply:
Fixed